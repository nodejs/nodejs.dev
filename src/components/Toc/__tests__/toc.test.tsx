import React from 'react';
import { render } from '@testing-library/react';
import TOC from '..';

describe('TOC component', () => {
  it('renders correctly', () => {
    const heading = 'TABLE OF CONTENTS';
    const tableOfContents = 'mock-toc';
    const { container } = render(
      <TOC heading={heading} tableOfContents={tableOfContents} />
    );
    expect(container).toMatchSnapshot();
  });
  it('render empty div without a table of contents', () => {
    const heading = 'TABLE OF CONTENTS';
    const tableOfContents = undefined;
    const { container } = render(
      <TOC heading={heading} tableOfContents={tableOfContents} />
    );
    expect(container).toMatchSnapshot();
  });
});
