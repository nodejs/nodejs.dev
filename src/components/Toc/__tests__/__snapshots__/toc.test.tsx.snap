// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOC component render empty div without a table of contents 1`] = `<div />`;

exports[`TOC component renders correctly 1`] = `
<div>
  <details
    class="toc"
  >
    <summary>
      <h6>
        TABLE OF CONTENTS
      </h6>
    </summary>
    <div>
      mock-toc
    </div>
  </details>
</div>
`;
